### PR TITLE
Add CSF pre and post workshop emails

### PR DESCRIPTION
**Pull request by rshipp**
Add new emails for CSF facilitators, that work in the same way as the matching emails for CSD facilitators.

Adds 4 new mailers:
* facilitator_pre_workshop_csf_intro
* facilitator_pre_workshop_csf_deepdive
* facilitator_post_workshop_csf_intro
* facilitator_post_workshop_csf_deepdive

It's hard to get a lot of confidence in these without having a way to test manually with real workshop data, so please do point out if there's any extra testing I can do here.

## Links

- jira ticket: [ACQ-750](https://codedotorg.atlassian.net/browse/ACQ-750)

### CODE REVIEW

File: dashboard/test/mailers/previews/pd_workshop_mailer_preview.rb [hunk 1]
```diff
@@ -165,6 +165,20 @@ def facilitator_pre_workshop_csd
...
+      Pd::Workshop::SUBJECT_CSF_101,
+      target: :facilitator
+  end
+
+  def facilitator_pre_workshop_csf_deepdive
+    mail :facilitator_pre_workshop,
+      Pd::Workshop::COURSE_CSF,
+      Pd::Workshop::SUBJECT_CSF_201,
+      target: :facilitator
+  end
```
Comment (davidsbailey) [inline dashboard/test/mailers/previews/pd_workshop_mailer_preview.rb:180]: you mentioned not having tested the emails, but I also see that you added previews. just checking whether you have viewed these at http://localhost-studio.code.org:3000/rails/mailers ?

File: dashboard/app/views/pd/workshop_mailer/facilitator_post_workshop.html.haml [hunk 2]
```diff
@@ -34,7 +34,10 @@ As a reminder, there are a number of ways for you to get support between worksho
   %li
     Engage with other facilitators in the
     = link_to('Facilitator Forum', 'https://forum.code.org/c/facilitators') + ' and'
-    = link_to('Facilitator Slack channel', 'https://codefacilitators6-12.slack.com/messages')
+    - if @workshop.course == 'CS Fundamentals'
+      = link_to('Facilitator Slack channel', 'https://csffacilitators.slack.com/messages')
+    - else
+      = link_to('Facilitator Slack channel', 'https://codefacilitators6-12.slack.com/messages')
```
Comment (TurnerRiley) [inline dashboard/app/views/pd/workshop_mailer/facilitator_post_workshop.html.haml:40]: @dmcavoy Did the url change for the Facilitator Slack channel or are there different urls between CSF and non-CSF? Just wondering if this if/else is needed or if we can do a single link?

File: dashboard/app/views/pd/workshop_mailer/facilitator_pre_workshop.html.haml [hunk 3]
```diff
@@ -18,9 +18,14 @@
...
       =link_to('facilitator landing page', 'https://code.org/educate/facilitator-landing/CSA') + '.'
+    - elsif @workshop.course == 'CS Fundamentals'
+      =link_to('facilitator landing page', 'https://code.org/educate/facilitator-landing/CSF') + '.'
   %li
     Have quick questions about resources, agendas, or participant requests? Post them in the facilitator
-    =link_to('Slack', 'https://codefacilitators6-12.slack.com/messages')
+    - if @workshop.course == 'CS Fundamentals'
+      =link_to('Slack', 'https://csffacilitators.slack.com/messages')
+    - else
+      =link_to('Slack', 'https://codefacilitators6-12.slack.com/messages')
```
Comment (TurnerRiley) [inline dashboard/app/views/pd/workshop_mailer/facilitator_pre_workshop.html.haml:28]: @dmcavoy Same question as above

File: dashboard/app/views/pd/workshop_mailer/facilitator_post_workshop.html.haml (hunk 2 above)
Comment (dmcavoy) [reply dashboard/app/views/pd/workshop_mailer/facilitator_post_workshop.html.haml:40]: There is a K-5 facilitator and 6-12 facilitator slack channel. So there are 2

File: dashboard/test/mailers/previews/pd_workshop_mailer_preview.rb (hunk 1 above)
Comment (rshipp) [reply dashboard/test/mailers/previews/pd_workshop_mailer_preview.rb:180]: yeah I have, and they work fine there! I also double checked the old csd previews to make sure I didn't break them

File: dashboard/app/views/pd/workshop_mailer/facilitator_pre_workshop.html.haml (hunk 3 above)
Comment (dmcavoy) [reply dashboard/app/views/pd/workshop_mailer/facilitator_pre_workshop.html.haml:28]: There is a K-5 facilitator and 6-12 facilitator slack channel. So there are 2

**Comment by davidsbailey** [comment]
> It's hard to get a lot of confidence in these without having a way to test manually with real workshop data, so please do point out if there's any extra testing I can do here.

to try to pull together a few of the things we've talked about regarding testing, I think you can effectively test this locally as follows:
1. set up local workshop + enrollee, either via web UI per Turner's steps or via command line (shared in #acquisition-products today)
2. tweak the workshop dates until it is N days from now (IIRC by changing the session dates, possible to do via command line as illustrated [here](https://docs.google.com/document/d/1d0oOA__XGomUDekWJd72mp4gUs6ID9JIXqTAFerskgU/edit#heading=h.jyzn72hg6bwo))
3. from local dashboard console, run `Pd::Workshop.send_automated_emails` 
4. confirm the email is logged to `dashboard/log/development.log`
5. repeat steps 2-4 for post surveys, setting workshop dates to N days ago and then running `Pd::Workshop.process_ends`

**Review by davidsbailey — APPROVED** [review]
code changes look good! I think this looks good to merge once you've verified locally that the right emails are getting sent per notes above.

having reviewed to the end of the PR, I realized that you may have to tweak some of the steps I wrote given that you are sending emails (in at least one case) to facilitators and not to participants. so, apologies if some of the details are off. I hope you get the idea! please feel free to ping me if you would like me to try and provide more specific/accurate steps.

**Comment by rshipp** [comment]
With Turner's docs and those commands, I was able to get this verified locally! thank you both

**Pull request merged**